### ISSUE TITLE
init --auto doesn't use btrfs storage by default on btrfs

### ISSUE THREAD

**Issue by albertodonato** [lxc/incus#379]
# Required information

 * Distribution: Ubuntu
 * Distribution version: 23.10
 * The output of "incus info" or if that fails:
   * Kernel version: `6.5.0-14-generic #14-Ubuntu SMP PREEMPT_DYNAMIC`
   * LXC version:
   * Incus version: 0.4
   * Storage backend in use: btrfs

# Issue description

On a system using btrfs as root (including `/var/lib/incus`), `incus admin init --auto` chooses the `dir` backend instead of `btrfs`.
In contrast, when doing an interactive `incus admin init`, `btrfs` is selected by default.

I'd be nice if `--auto` would also select btrfs by default when a btrfs filesystem is detected.

**Comment by freeekanayaka** [comment]
Hey Alberto, nice to see you around here ;)

**Comment by stgraber** [comment]
Yeah, `--auto` is pretty dumb currently, it always uses the `dir` storage backend regardless of whatever else may be available. I'll have to see if we can make it a bit smarter without breaking stuff.
